lib/fog/core/connection.rb

```diff
@@ -77,6 +75,13 @@ def reset
 
       private
 
+      def user_agent
+        arr = [ENV['FOG_CUSTOM_USER_AGENT']] # Allow apps to pass custom UA for metrics, etc.
```

Comment:
This should probably also use an `if` instead of just adding nil (otherwise when you join you end up with leading whitespace, which is why the tests break).